src/contract.ts

```diff
@@ -0,0 +1,381 @@
+import {
+  Keypair,
+  PublicKey,
+  TransactionInstruction,
+  SYSVAR_CLOCK_PUBKEY,
+} from '@solana/web3.js';
+import { ethers } from 'ethers';
+
+import { EventCallback, parseLogTopic } from './logs';
+import { encodeSeeds, numToPaddedHex } from './utils';
+import { Program } from './program';
+
+export type ContractFunction<T = any> = (...args: Array<any>) => Promise<T>;
+
+export type ContractDeployOptions = {
+  name: string;
+  abi: string;
+  space: number;
+  constructorArgs?: any[];
+  accounts?: PublicKey[];
+  writableAccounts?: PublicKey[];
+  seeds?: any[];
+  signers?: Keypair[];
+  caller?: PublicKey | undefined;
+  value?: number;
+  simulate?: boolean;
+};
+
+export type ContractTransactionOptions = {
+  accounts?: PublicKey[];
+  writableAccounts?: PublicKey[];
+  seeds?: any[];
+  signers?: Keypair[];
+  caller?: PublicKey | undefined;
+  value?: number;
+  simulate?: boolean;
+};
```

Comment:
Could these `any` types be more strict? For function and constructor args especially, it would be helpful to have inputs that can be generically typed and known to the caller.

Reply:
The seeds need types for sure. If the args/constructor args can be more restricted, that would be great.

Reply:
Fixed.